vm/src/stdlib/os.rs

```diff
@@ -2486,10 +2486,140 @@ mod posix {
         }
     }
 
+    #[derive(FromArgs)]
+    pub struct SchedParamArg {
+        #[pyarg(any)]
+        sched_priority: PyObjectRef,
+    }
+
     impl SlotConstructor for SchedParam {
-        type Args = SchedParam;
-        fn py_new(cls: PyTypeRef, sched_param: Self::Args, vm: &VirtualMachine) -> PyResult {
-            sched_param.into_pyresult_with_type(vm, cls)
+        type Args = SchedParamArg;
+        fn py_new(cls: PyTypeRef, arg: Self::Args, vm: &VirtualMachine) -> PyResult {
+            SchedParam {
+                sched_priority: arg.sched_priority,
+            }
+            .into_pyresult_with_type(vm, cls)
+        }
+    }
+
+    #[derive(FromArgs)]
+    struct SetSchedArgs {
+        #[pyarg(positional)]
+        pid: i32,
+        #[pyarg(positional)]
+        policy: i32,
+        #[pyarg(positional)]
+        sched_param_obj: PyRef<SchedParam>,
+    }
+
+    #[cfg(any(
+        target_os = "linux",
+        target_os = "netbsd",
+        target_os = "freebsd",
+        target_os = "android"
+    ))]
+    #[pyfunction]
+    fn sched_getscheduler(pid: libc::pid_t, vm: &VirtualMachine) -> PyResult<i32> {
+        let policy = unsafe { libc::sched_getscheduler(pid) };
+        if policy == -1 {
+            Err(errno_err(vm))
+        } else {
+            Ok(policy)
+        }
+    }
+
+    #[cfg(any(
+        target_os = "linux",
+        target_os = "netbsd",
+        target_os = "freebsd",
+        target_os = "android"
+    ))]
+    fn convert_sched_param(
```

Comment:
this is a converter from `SchedParam` to `libc::sched_param`. I think this is a method rather than a free function. I think `SchedParam::try_to_libc(vm)` fits.
Then it will be used like:
```rust
let libc_sched_param = args.sched_param_obj.try_to_libc(vm)?;
```